### ISSUE TITLE
ORA-00972 When migrate to 24.x.x

### ISSUE THREAD

**Issue by ZouhaierSebri** [keycloak/keycloak#28371]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

core

### Describe the bug

2024-04-02 16:02:48,514 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: Failed to start server in (development) mode
2024-04-02 16:02:48,514 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: Failed to update database
2024-04-02 16:02:48,514 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for changeset META-INF/jpa-changelog-24.0.0.xml::24.0.0-9758::keycloak:
     Reason: liquibase.exception.DatabaseException: ORA-00972: identifier is too long

### Version

24.0.2

### Regression

- [ ] The issue is a regression

### Expected behavior

Use INDEX  names with length <= 30  for Oracle

### Actual behavior

Keycloak not started on Oracle DB

### How to Reproduce?

just use Oracle DB for keycloak

**Comment by vramik** [comment]
@ZouhaierSebri thank you for reporting the issue. I believe it is duplicate of #27590. Please take a look there for more context. 

TL;DR Since Oracle 12.2 there is support for longer than 30 identifiers. Don't you have set a compatibility mode?
